apps/designer/app/designer/features/style-panel/sections/spacing/keyboard.ts

```diff
@@ -0,0 +1,102 @@
+import { useState } from "react";
+import type { FocusEvent, KeyboardEvent, MouseEvent } from "react";
+import type { SpacingStyleProperty } from "./types";
+
+const movementKeys = [
+  "ArrowUp",
+  "ArrowRight",
+  "ArrowDown",
+  "ArrowLeft",
+] as const;
+
+const movementMap = {
+  marginTop: ["marginBottom", "marginRight", "paddingTop", "marginLeft"],
+  marginRight: ["marginTop", "marginLeft", "marginBottom", "paddingRight"],
+  marginBottom: ["paddingBottom", "marginRight", "marginTop", "marginLeft"],
+  marginLeft: ["marginTop", "paddingLeft", "marginBottom", "marginRight"],
+  paddingTop: ["marginTop", "paddingRight", "paddingBottom", "paddingLeft"],
+  paddingRight: ["paddingTop", "marginRight", "paddingBottom", "paddingBottom"],
+  paddingBottom: ["paddingTop", "paddingRight", "marginBottom", "paddingLeft"],
+  paddingLeft: ["paddingTop", "paddingTop", "paddingBottom", "marginLeft"],
+} as const;
+
+export const useKeyboardNavigation = ({
```

Comment:
do you see any way to reuse the arrow keys logic with other components? e.g. flex control

Reply:
Not sure I understand. Do we need the same logic there? If so, I would copy paste at first, make all necessary changes there, and then would consider merging two similar components into one reusable.

Reply:
The same thing in all complex components is
1. tab to focus the most parent element
2. use arrow keys to navigate internal components

This is gonna be the same for all complex components

Reply:
Still would think about how to extract after we have at least two instances

Reply:
Problem is that its unlikely anybody will touch this logic if its not designed to be reusable, because they would have to test both components and will be afraid to break them, so we will unlikely reuse the logic if we don't make it reusable right away.

On the other hand, its difficult to make a good abstraction without actually reusing it, so maybe we should just do it with flex now so that this can be refactored into a reusable piece as long as its still fresh

Reply:
> Problem is that its unlikely anybody will touch this logic if its not designed to be reusable, because they would have to test both components and will be afraid to break them, so we will unlikely reuse the logic if we don't make it reusable right away.

That's solvable by either putting the same person (me) on implementing this in other components. Or by making a note in the issue about implementing this in other component.

> On the other hand, its difficult to make a good abstraction without actually reusing it

Yeah, that's basically my concern. You end up adding unnecessary complexity by making some parts unnecessarely flexible, at the same time you don't make other parts flexible enough.

Also, sometimes two simple components are easier to maintain than one reusable.

> so maybe we should just do it with flex now so that this can be refactored into a reusable piece as long as its still fresh

I'll create an issue